Userland/Libraries/LibJS/Runtime/Temporal/ZonedDateTime.cpp

```diff
@@ -554,7 +554,12 @@ ThrowCompletionOr<NanosecondsToDaysResult> nanoseconds_to_days(VM& vm, Crypto::S
     if (!nanoseconds.is_negative() && sign == -1)
         return vm.throw_completion<RangeError>(ErrorType::TemporalInvalidEpochNanoseconds);
 
-    // 23. Return the Record { [[Days]]: days, [[Nanoseconds]]: nanoseconds, [[DayLength]]: abs(dayLengthNs) }.
+    // 23. If abs(nanoseconds) ≥ abs(dayLengthNs), throw a RangeError exception.
+    auto compare_result = nanoseconds.compare_to_double(fabs(day_length_ns.to_double()));
```

Comment:
Unless I'm missing something here you don't do the abs part of `abs(nanoseconds)`.

Reply:
I see that `UnsignedBigInteger` doesn't have a `compare_to_double` function, so just matching the sign of `nanoseconds might be easier? Or you could temporarily negate nanoseconds

Reply:
I will go with the negation, and maybe in the future will add compare_to_double() to UnsignedBigInteger.